Mapsui.Geometries/ArrowLine.cs

```diff
@@ -0,0 +1,196 @@
+        {
+            get
+            {
+                return new Point((StartPoint.X + EndPoint.X) / 2, (StartPoint.Y + EndPoint.Y) / 2);
+            }
+        }
+
+        /// <summary>
+        ///     The two points linked to the 'ArrowHead' in order to draw the sides of the arrow
+        /// </summary>
+        /// <param name="radiansAngle"></param>   
+        /// <returns>Returns a set of two Points</returns>
+        public Point[] ArrowExtremities(double radiansAngle, double length)
+        {
+            if (IsEmpty())
+                return new Point[] { ArrowHead, ArrowHead };
+
+            double slope;
+            if (StartPoint.X == EndPoint.X)  //Prevents div/0
+                slope = double.PositiveInfinity;
+            else
+                slope = (StartPoint.Y - EndPoint.Y) / (StartPoint.X - EndPoint.X);
+
+            double arrowOrientation = (StartPoint.X - EndPoint.X) / Math.Abs(StartPoint.X - EndPoint.X);
+
+            Point firstExt = new Point(ArrowHead.X + arrowOrientation * length * Math.Cos(Math.Atan(slope) - radiansAngle), ArrowHead.Y + arrowOrientation * length * Math.Sin(Math.Atan(slope) - radiansAngle));
+            Point secondExt = new Point(ArrowHead.X + arrowOrientation * length * Math.Cos(Math.Atan(slope) + radiansAngle), ArrowHead.Y + arrowOrientation * length * Math.Sin(Math.Atan(slope) + radiansAngle));
+
+            return new Point[] { firstExt, secondExt };
+        }
+
+        /// <summary>
+        ///     The minimum bounding box for this Geometry.
+        /// </summary>
+        /// <returns>BoundingBox for this geometry</returns>
+        public override BoundingBox GetBoundingBox()
+        {
+            if (IsEmpty())
+                return null;
+            Point[] pointsCollection = new Point[] { StartPoint, EndPoint, ArrowExtremities(RadiansAngle, ArrowLength)[0], ArrowExtremities(RadiansAngle, ArrowLength)[1] };
+            var bbox = new BoundingBox(pointsCollection[0], pointsCollection[0]);
+            for (var i = 1; i < 4; i++)
+            {
+                bbox.Min.X = pointsCollection[i].X < bbox.Min.X ? pointsCollection[i].X : bbox.Min.X;
+                bbox.Min.Y = pointsCollection[i].Y < bbox.Min.Y ? pointsCollection[i].Y : bbox.Min.Y;
+                bbox.Max.X = pointsCollection[i].X > bbox.Max.X ? pointsCollection[i].X : bbox.Max.X;
+                bbox.Max.Y = pointsCollection[i].Y > bbox.Max.Y ? pointsCollection[i].Y : bbox.Max.Y;
+            }
+            return bbox;
+        }
+
+        /// <summary>
+        ///     If true, then this Geometry represents the empty point set, Ø, for the coordinate space.
+        /// </summary>
+        /// <returns>Returns 'true' if this Geometry is the empty geometry</returns>
+        public override bool IsEmpty()
+        {
+            return StartPoint.Equals(EndPoint);
+        }
+
+        public override bool Contains(Point point)
+        {
+            return Distance(point) == 0;
```

Comment:
Not sure this is the right answer, see LineString.Contains implementation.

Reply:
LineString.Contains implementation seems to be unused since it just returns 'false'. My idea was to say " if the distance between a shape and a point is equal to 0, then the point is somewhere on the shape, so the shape contains the point ". What other explanation do you see for 'Contains' ?